### PR TITLE
fragroute: fix compile against newer libpcap versions

### PR DESCRIPTION
**Pull request by mitchblank**
There isn't really a place to upstream patches for this; there are a some unofficial github repos but I think the last official release from @dugsong was circa 2002.

Still, with a little bit more patching it does compile OK on MacOS 11.

I think an `inreplace` is the simplest workaround in this case.  Afterwards I did some searching and found other places that had done equivalent patches like https://gitweb.gentoo.org/repo/gentoo.git/tree/net-analyzer/fragroute/files/fragroute-1.2.6-pcap_open.patch although Gentoo is tracking the "fragroute-ipv6" upstream which seems to be another fork of the same project (see https://github.com/stsi/fragroute-ipv6 )   Even `fragroute-ipv6` saw its latest tarball release in 2012... the github has some activity as recently as this year but would also need a similar change to work with recent libpcap (cc @stsi )

For reference, the `pcap_open()` API that is causing the conflict was added in 2017 ( https://github.com/the-tcpdump-group/libpcap/commit/a372536befc9b1ee1d355058daa2eb9f66aa0c59 ) but probably only made it to the MacOS system headers with Big Sur

### CODE REVIEW

**Comment by SMillerDev** [comment]
Seeing how Gentoo seems to be the only big package manager shipping this, I'd say it's fine to switch to the fork to at least get _something_ newer.

**Comment by mitchblank** [comment]
> Seeing how Gentoo seems to be the only big package manager shipping this, I'd say it's fine to switch to the fork to at least get _something_ newer.

No complaints, but maybe we can merge this as an intermediate step to get the current version compiling and then do the switch as a followup PR?  That way we've got a bit more history maintained in case we have to switch back for some reason.

**Comment by SMillerDev** [comment]
> No complaints, but maybe we can merge this as an intermediate step to get the current version compiling and then do the switch as a followup PR? That way we've got a bit more history maintained in case we have to switch back for some reason.

Sure, just waiting for CI

**Comment by stsi** [comment]
Thanks for the notification. I've fixed fragroute-ipv6 in case you want to use it.
See https://github.com/stsi/fragroute-ipv6/commit/f16f6ac0288a8814461b6e5f2df0fe455f887433

BTW, the original dugsong's fragroute project is dead for all practical purposes. I don't want to claim that fragroute-ipv6 is the "official" fragroute, but I do respond to emails and fix small issues.

**Review by SMillerDev — APPROVED** [review]
Thanks @mitchblank  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by mitchblank** [comment]
Did a little bit more digging -- Gentoo switched to tracking the "fragroute-ipv6" fork on 2012-10-11 https://gitweb.gentoo.org/repo/gentoo/historical.git/commit/net-analyzer/fragroute?id=b4799d87ae84750f81cf378009f770de0e90e78e

Was hoping that there would be more discussion there about the decision but didn't find much.  Two days earlier the same committer had updated with some new patches but was still tracking the monkey.org original but then decided to track -ipv6 instead

**Comment by mitchblank** [comment]
OK, now that it's merged I'll try to make a PR to track fragroute-ipv6 instead

**Comment by mitchblank** [comment]
I messed with fragroute-ipv6 for awhile... it's a bit of a chore

It seems that it heavily relies on its own fork of libdnet or you get:
```
fragroute.c:33:2: error: unknown type name 'ndisc_t'
        ndisc_t         *ndisc;
        ^
```
We have a bunch of formula that rely on libdnet so probably moving *that* formula to a fork as well would not be wise.

@stsi has a libdnet patch at https://github.com/stsi/fragroute-ipv6/issues/2 and it seems to have been upstreamed at https://github.com/ofalk/libdnet/pull/56 but I don't know if we should wait for the next official libdnet and try again or if we should take that libdnet patch into homebrew early...

**Pull request closed without merging**
